src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -39,8 +43,36 @@ public override bool Execute()
                 SleepTimeInMilliseconds = s_DefaultSleepTime;
             }
 
-            return CleanupAgentsAsync().Result;
+            bool returnValue = true;
+
+            if(Report)
+            {
+                ReportDiskUsage();
+            }
+            if(Clean)
+            {
+                returnValue &= CleanupAgentsAsync().Result;
+            }
+            // If report and clean are both 'true', then report disk usage both before and after cleanup.
+            if(Report && Clean)
+            {
+                ReportDiskUsage();
+            }
+
+            return returnValue;
         }
+
+        private void ReportDiskUsage()
+        {
+            string drive = Path.GetPathRoot(AgentDirectory);
+            DriveInfo driveInfo = new DriveInfo(drive);
+            Console.WriteLine("Disk Usage Report");
+            Console.WriteLine($"  Agent directory: {AgentDirectory}");
```

Comment:
I think it would be interesting to get the size and number of agent directories as well.